File: 3263-divide-an-array-into-subarrays-with-minimum-cost-i/divide-an-array-into-subarrays-with-minimum-cost-i.java
class Solution {
    public int minimumCost(int[] nums) {
        int n = nums.length;

        int Min1 = 51, Min2 = 51;
        for (int i = 1; i < n; i++) {
            if (nums[i] <= Min1) {
                Min2 = Min1;
                Min1 = nums[i];
            } else if (nums[i] < Min2) {
                Min2 = nums[i];
            }
        }

        return Min1 + Min2 + nums[0];
    }
}

File: 3263-divide-an-array-into-subarrays-with-minimum-cost-i/divide-an-array-into-subarrays-with-minimum-cost-i.cpp
class Solution {
public:
    int minimumCost(vector<int>& nums) {
        int n = nums.size();

        priority_queue<int> pq;
        for (int i = 1; i < n; i++) {
            if (pq.size() == 2) {
                if (pq.top() > nums[i]) {
                    pq.pop();
                    pq.push(nums[i]);
                }
            } else {
                pq.push(nums[i]);
            }
        }

        int result = nums[0] + pq.top(); pq.pop();
        return result + pq.top();
    }
};